src/app/OperationalDeviceProxy.cpp

```diff
@@ -98,6 +98,10 @@ CHIP_ERROR OperationalDeviceProxy::Connect(Callback::Callback<OnDeviceConnected>
     if (err != CHIP_NO_ERROR && onFailure != nullptr)
     {
         onFailure->mCall(onFailure->mContext, mPeerId, err);
+
+        DequeueConnectionSuccessCallbacks(/* executeCallback */ false);
+        DequeueConnectionFailureCallbacks(err, /* executeCallback */ true);
+        CloseCASESession();
```

Comment:
This should really live inside `OperationalDeviceProxy::EstablishConnection` in the case when `EstablishSession` returns error, because there it's clear we need to clean up mCASEClient....  Putting it here is a little odd.

Reply:
https://github.com/project-chip/connectedhomeip/pull/17466 addresses this.